client/extensions/woocommerce/state/settings/shipping/zones/reducer.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * External dependencies
+ */
+import { isEqual, unionWith, differenceWith, isEmpty } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import {
+	WOOCOMMERCE_SHIPPING_ZONE_ADD,
+	WOOCOMMERCE_SHIPPING_ZONE_CANCEL,
+	WOOCOMMERCE_SHIPPING_ZONE_CLOSE,
+	WOOCOMMERCE_SHIPPING_ZONE_EDIT,
+	WOOCOMMERCE_SHIPPING_ZONE_LOCATION_ADD,
+	WOOCOMMERCE_SHIPPING_ZONE_LOCATION_REMOVE,
+	WOOCOMMERCE_SHIPPING_ZONE_METHOD_ADD,
+	WOOCOMMERCE_SHIPPING_ZONE_METHOD_CHANGE_TYPE,
+	WOOCOMMERCE_SHIPPING_ZONE_METHOD_EDIT,
+	WOOCOMMERCE_SHIPPING_ZONE_METHOD_REMOVE,
+	WOOCOMMERCE_SHIPPING_ZONE_REMOVE,
+} from '../../../action-types';
+
+const LOCATION_TYPES = [ 'postcode', 'state', 'country', 'continent' ];
+
+// TODO: reorder zones (do we need it for MVP?)
+
+const reducers = {};
+
+reducers[ WOOCOMMERCE_SHIPPING_ZONE_ADD ] = ( state ) => {
+	return { ...state,
+		currentlyEditingZone: {
+			id: null,
+			locations: [],
+			methods: [],
+			order: ( ( state.zones[ state.zones.length - 2 ] || {} ).order || 0 ) + 1,
+		},
+		currentlyEditingZoneIndex: -1,
+	};
+};
+
+reducers[ WOOCOMMERCE_SHIPPING_ZONE_EDIT ] = ( state, { index } ) => {
+	return { ...state,
+		currentlyEditingZone: state.zones[ index ],
+		currentlyEditingZoneIndex: index,
+	};
+};
+
+reducers[ WOOCOMMERCE_SHIPPING_ZONE_REMOVE ] = ( state, { index } ) => {
+	// Protect { id: 0 } (Rest of the World) from deletion
+	if ( 0 === state.zones[ index ].id ) {
```

Comment:
If we aren't going to allow the "Rest of the World" zone to be reordered, or even hiding the zone completely, why not handle it separately in state?